src/core/services/compiler/compiler.js

```diff
@@ -248,13 +266,8 @@ MdCompilerService.prototype._compileElement = function(locals, element, options)
         $element: element
       });
 
-      var invokeCtrl = self.$controller(options.controller, injectLocals, true, options.controllerAs);
-
-      if (options.bindToController) {
-        angular.extend(invokeCtrl.instance, locals);
-      }
-
-      var ctrl = invokeCtrl();
+      // Create the specified controller instance.
+      var ctrl = self._createController(options, injectLocals, locals);
```

Comment:
In the way that `angular.extend(locals, {$element: ...})` extends `locals` with `$element` and then returns `locals`. So, `injectLocals === locals`. I assume the intended behavior was `injectLocals = angular.extend({$element: ...}, locals)` or `injectLocals = angular.extend({}, locals, {$element: ...})`.